src/core/providers/ept/qgseptdecoder.cpp

```diff
@@ -121,37 +121,67 @@ bool _serialize( QByteArray &data, size_t outputPosition, const QgsPointCloudAtt
 
 QgsPointCloudBlock *_decompressBinary( const QByteArray &dataUncompressed, const QgsPointCloudAttributeCollection &attributes, const QgsPointCloudAttributeCollection &requestedAttributes )
 {
-  const int pointRecordSize = attributes.pointRecordSize( );
-  const int requestedPointRecordSize = requestedAttributes.pointRecordSize();
+  const std::size_t pointRecordSize = attributes.pointRecordSize( );
+  const std::size_t requestedPointRecordSize = requestedAttributes.pointRecordSize();
   const int count = dataUncompressed.size() / pointRecordSize;
   QByteArray data;
   data.resize( requestedPointRecordSize * count );
+  char *destinationBuffer = data.data();
   const char *s = dataUncompressed.data();
 
+  const QVector<QgsPointCloudAttribute> requestedAttributesVector = requestedAttributes.attributes();
+
+  // calculate input attributes and offsets once in advance
+
+  struct AttributeData
+  {
+    AttributeData( int inputOffset, int inputSize, QgsPointCloudAttribute::DataType inputType, int requestedSize, QgsPointCloudAttribute::DataType requestedType )
+      : inputOffset( inputOffset )
+      , inputSize( inputSize )
+      , inputType( inputType )
+      , requestedSize( requestedSize )
+      , requestedType( requestedType )
+    {}
+
+    int inputOffset;
+    int inputSize;
+    QgsPointCloudAttribute::DataType inputType;
+    int requestedSize;
+    QgsPointCloudAttribute::DataType requestedType;
+  };
+
+  std::vector< AttributeData > attributeData;
```

Comment:
qvector?

Reply:
Since it's only internal, I trust the compiler here to better optimise this vs Qt :laugh: . Also means we don't incur the cost of some qt container checks which are expensive on debug builds. They are skipped/optimised out on release builds, but I almost always use debug builds and want faster stuff anyway!

Reply:
micro-optimization? :)

Reply:
Maybe... but it **did** make a difference in my test builds!

Read the upstream discussion around the qt containers. They are disliked by the qt developers and I gather that they'd happily remove them all and force everyone to use std containers.

I'm 50/50 on this. I prefer std containers wherever possible, but the qt containers have two nice things which we use in qgis a lot:
1. implicit sharing
2. sip/python compatibility

That's why my general rule of thumb is (in the absence of all other factors) use std:: for private stuff, use Qt for public stuff.

(lastly -- qhash is often better then the std container, and will be even better on qt 6 builds. There's no easy answers!)